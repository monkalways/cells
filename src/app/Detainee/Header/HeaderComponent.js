import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar,
  Grid,
  IconButton,
  Toolbar,
  withStyles,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

import logoIcon from './eps-logo-icon.png';
import HeaderContent from './HeaderContent';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({}),
  isDetaineeProfileLoaded: PropTypes.bool.isRequired,
  onLogout: PropTypes.func.isRequired,
};

const defaultProps = {
  detainee: null,
};

const Header = ({
  classes, detainee, isDetaineeProfileLoaded, onLogout,
}) => (
  <div className={classes.root}>
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <IconButton
          className={classes.menuButton}
          color="inherit"
          aria-label="Menu"
        >
          <img src={logoIcon} alt="EPS Logo" width={70} />
        </IconButton>
        {isDetaineeProfileLoaded ? (
          <HeaderContent
            detainee={detainee}
            isDetaineeProfileLoaded={isDetaineeProfileLoaded}
          />
        ) : (
          <Grid item xs={12} />
        )}
        <IconButton
          className={classes.logoutButton}
          color="inherit"
          aria-label="Menu"
          onClick={onLogout}
        >
          <CloseIcon className={classes.logoutIcon} />
        </IconButton>
      </Toolbar>
    </AppBar>
  </div>
);

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default withStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: '#A8C6FA', // TODO: move color to theme
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  logoutButton: {
    margin: theme.spacing.unit,
    backgroundColor: '#444480',
    color: '#F10607',
    '&:hover': {
      backgroundColor: '#444480',
    },
  },
  logoutIcon: {
    fontSize: theme.typography.h4.fontSize,
  },
}))(Header);
