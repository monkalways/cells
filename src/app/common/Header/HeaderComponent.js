import React from 'react';
import PropTypes from 'prop-types';
import {
  AppBar, Toolbar, Typography, withStyles,
} from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  versions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
  })).isRequired,
};

export const HeaderComponent = ({ classes, versions }) => (
  <AppBar position="static">
    <Toolbar>
      <Typography
        id="title"
        variant="h6"
        color="inherit"
        className={classes.heading}
      >
        Cell App - Detainee Management Unit
      </Typography>
      <div>
        {versions
          && versions.map((version) => (
            <Typography
              id="version"
              color="inherit"
              align="right"
              key={version.name}
            >
              {version.name}: {version.version}
            </Typography>
          ))}
      </div>
    </Toolbar>
  </AppBar>
);

HeaderComponent.propTypes = propTypes;

export default withStyles(() => ({
  heading: {
    flex: 1,
  },
}))(HeaderComponent);
