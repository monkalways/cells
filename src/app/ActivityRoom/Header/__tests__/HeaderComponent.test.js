import React from 'react';
import { shallow } from 'enzyme';

import { IconButton } from '@material-ui/core';
import { HeaderComponent } from '../HeaderComponent';

describe('HeaderComponent', () => {
  let usage;
  let onLogout;
  let isAuthenticated;

  beforeEach(() => {
    usage = 'phone';
    onLogout = jest.fn();
    isAuthenticated = false;
  });

  const setup = () => {
    const classes = {
      root: {},
      appBar: {},
      menuButton: {},
      logoutButton: {},
    };
    return shallow(<HeaderComponent
      classes={classes}
      usage={usage}
      onLogout={onLogout}
      isAuthenticated={isAuthenticated}
    />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render logout button if authenticated', () => {
    isAuthenticated = true;
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(IconButton)).toExist();
  });
});
