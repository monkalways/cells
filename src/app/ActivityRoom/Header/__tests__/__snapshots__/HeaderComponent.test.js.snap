// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render 1`] = `
<div
  className=""
>
  <WithStyles(AppBar)
    className=""
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        className=""
        color="inherit"
      >
        <img
          alt="EPS Logo"
          src="eps-logo-icon.png"
          width={70}
        />
      </WithStyles(IconButton)>
      <WithStyles(HeaderContentComponent)
        usage="phone"
      />
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`HeaderComponent should render logout button if authenticated 1`] = `
<div
  className=""
>
  <WithStyles(AppBar)
    className=""
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Menu"
        className=""
        color="inherit"
      >
        <img
          alt="EPS Logo"
          src="eps-logo-icon.png"
          width={70}
        />
      </WithStyles(IconButton)>
      <WithStyles(HeaderContentComponent)
        usage="phone"
      />
      <WithStyles(IconButton)
        aria-label="Menu"
        className=""
        color="inherit"
        onClick={[MockFunction]}
      >
        <pure(CloseIcon) />
      </WithStyles(IconButton)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
